hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/server/HermesServer.java

```diff
@@ -97,9 +97,18 @@ public void start() {
         if (configFactory.getBooleanProperty(FRONTEND_TOPIC_METADATA_REFRESH_JOB_ENABLED)) {
             topicMetadataLoadingJob.start();
         }
+        healthCheckService.startup();
         readinessChecker.start();
     }
 
+    public void stop() throws InterruptedException {
+        boolean isGraceful = configFactory.getBooleanProperty(FRONTEND_GRACEFUL_SHUTDOWN_ENABLED);
+        if(isGraceful) {
+            gracefulShutdown();
+        }
+        shutdown();
```

Comment:
Yes, because there is completely different logic hidden underneath this two functions, I renamed `gracefulShutdown()` to `gracefullyPrepareForShutdown()` to be less confusing